src/main/java/net/openhft/chronicle/core/util/ThreadIndexAssigner.java

```diff
@@ -44,6 +44,6 @@ public int getId() throws IllegalStateException, BufferOverflowException, Buffer
     }
 
     protected int nextIndex(int size) {
-        return ThreadLocalRandom.current().nextInt(size);
+        return new SecureRandom().nextInt(size);
```

Comment:
Is it worth implementing a ThreadLocalSecureRandom (similar to ThreadLocalRandom), this seems like a sonar warning that will come up again and again? Also Core might be a good place to put it.

Reply:
Or shall we just suppress the sonar warning? I don't see it provides much benefit - we don't use randoms for anything important, or attackable

Reply:
In this case, there is no performance issue so I think we can stick with SecureRandom